PathPlanning/FlowField/flowfield.py

```diff
@@ -0,0 +1,214 @@
+        the lowest cost in the integration field. An agent will simply
+        follow this vector field to the goal"""
+        self.vector_field = {}
+        for i in range(self.limit_x):
+            for j in range(self.limit_y):
+                if self.obs_grid[(i, j)] == 'obs':
+                    continue
+                if [i, j] == self.goal_pt:
+                    self.vector_field[(i, j)] = (None, None)
+                    continue
+                offset_list = [(i + a, j + b)
+                               for a in range(-1, 2)
+                               for b in range(-1, 2)]
+                neighbor_list = [{'loc': pt,
+                                  'cost': self.integration_field[pt]}
+                                 for pt in offset_list
+                                 if self.obs_grid[pt] != 'obs']
+                neighbor_list = sorted(neighbor_list, key=lambda x: x['cost'])
+                best_neighbor = neighbor_list[0]['loc']
+                self.vector_field[(i, j)] = best_neighbor
+
+    def follow_vectors(self):
+        curr_x, curr_y = self.start_pt
+        while curr_x is not None and curr_y is not None:
+            plt.plot(curr_x, curr_y, "b*")
+            curr_x, curr_y = self.vector_field[(curr_x, curr_y)]
+            plt.pause(0.001)
+        if show_animation:
+            plt.show()
+
+
+def main():
+    # set obstacle positions
+    obs_dict = {}
+    for i in range(51):
+        for j in range(51):
+            obs_dict[(i, j)] = 'free'
+    o_x, o_y, m_x, m_y, h_x, h_y = [], [], [], [], [], []
+
+    s_x = 5.0
+    s_y = 5.0
+    g_x = 35.0
+    g_y = 45.0
+
+    # draw outer border of maze
+    draw_vertical_line(0, 0, 50, o_x, o_y, obs_dict, 'obs')
+    draw_vertical_line(48, 0, 50, o_x, o_y, obs_dict, 'obs')
+    draw_horizontal_line(0, 0, 50, o_x, o_y, obs_dict, 'obs')
+    draw_horizontal_line(0, 48, 50, o_x, o_y, obs_dict, 'obs')
+
+    # draw inner walls
+    all_x = [10, 10, 10, 15, 20, 20, 30, 30, 35, 30, 40, 45]
+    all_y = [10, 30, 45, 20, 5, 40, 10, 40, 5, 40, 10, 25]
+    all_len = [10, 10, 5, 10, 10, 5, 20, 10, 25, 10, 35, 15]
+    for x, y, l in zip(all_x, all_y, all_len):
+        draw_vertical_line(x, y, l, o_x, o_y, obs_dict, 'obs')
+
+    all_x[:], all_y[:], all_len[:] = [], [], []
+    all_x = [35, 40, 15, 10, 45, 20, 10, 15, 25, 45, 10, 30, 10, 40]
+    all_y = [5, 10, 15, 20, 20, 25, 30, 35, 35, 35, 40, 40, 45, 45]
+    all_len = [10, 5, 10, 10, 5, 5, 10, 5, 10, 5, 10, 5, 5, 5]
+    for x, y, l in zip(all_x, all_y, all_len):
+        draw_horizontal_line(x, y, l, o_x, o_y, obs_dict, 'obs')
+
+    # draw paths that have medium difficulty (in terms of going through them)
+    all_x[:], all_y[:], all_len[:] = [], [], []
+    all_x = [10, 45]
+    all_y = [22, 20]
+    all_len = [8, 5]
+    for x, y, l in zip(all_x, all_y, all_len):
+        draw_vertical_line(x, y, l, m_x, m_y, obs_dict, 'medium')
```

Comment:
What do you mean the "medium" and "hard"? Could you please add comment to explain it in here or header comment?